source/parse.go

```diff
@@ -19,10 +19,13 @@ type Entry map[string]any
 // It also supports multidoc YAML, and will either return the root object itself if that
 // root is a map[string]any, or if the root is an array, will try returning the contents
 // of said array that are map[string]any's.
-func Parse(filename string, data []byte) []Entry {
+func Parse(filename string, data []byte) ([]Entry, error) {
 	// Try Jsonnet first, which will also cover JSON.
-	jsonString, err := jsonnet.MakeVM().EvaluateSnippet(filename, string(data))
-	if err == nil {
+	if jsonString, err := jsonnet.MakeVM().EvaluateSnippet(filename, string(data)); err != nil {
+		if strings.HasPrefix(err.Error(), "RUNTIME ERROR") {
+			return []Entry{}, err
+		}
+	} else {
 		data = []byte(jsonString)
 	}
```

Comment:
It would be a little neater (and more idiomatic) to do something like:

```
var jsonnetError jsonnet.RuntimeError
if errors.As(err, &jsonnetError) {
	// handle error
}
```

This code also swallows any other errors, and so I think we need to work out an approach for those cases.